### PR TITLE
KFSPTS-10646: Add Fields to WebServiceCredential

### PR DESCRIPTION
**Pull request by ajd299**
https://confluence.cornell.edu/pages/viewpage.action?pageId=352652470
https://jira.cornell.edu/browse/KFSPTS-10646

Add Credential Group, Active Indicator and Last Updated Timestamp fields to the Object and update usage of object to include Credential Group

Note these changes have associated SQL Schema changes

### CODE REVIEW

File: src/main/java/edu/cornell/kfs/sys/businessobject/WebServiceCredential.java [hunk 1]
```diff
@@ -1,11 +1,22 @@
 package edu.cornell.kfs.sys.businessobject;
 
 import org.kuali.kfs.krad.bo.PersistableBusinessObjectBase;
+import org.kuali.rice.core.api.mo.common.active.MutableInactivatable;
 
-public class WebServiceCredential extends PersistableBusinessObjectBase{
+public class WebServiceCredential extends PersistableBusinessObjectBase implements MutableInactivatable{
```
Comment (cah292) [inline src/main/java/edu/cornell/kfs/sys/businessobject/WebServiceCredential.java]: Missing space before the curly brace.

File: src/main/java/edu/cornell/kfs/sys/businessobject/WebServiceCredential.java [hunk 2]
```diff
@@ -21,4 +32,12 @@ public String getCredentialValue() {
     public void setCredentialValue(String credentialValue) {
         this.credentialValue = credentialValue;
     }
+
+    public boolean isActive(){
```
Comment (cah292) [inline src/main/java/edu/cornell/kfs/sys/businessobject/WebServiceCredential.java]: Another missing space before the curly brace.

File: src/main/java/edu/cornell/kfs/sys/businessobject/WebServiceCredential.java [hunk 3]
```diff
@@ -21,4 +32,12 @@ public String getCredentialValue() {
...
+        return active;
+    }
+
+    public void setActive(boolean active){
```
Comment (cah292) [inline src/main/java/edu/cornell/kfs/sys/businessobject/WebServiceCredential.java]: Another missing space before the curly brace.

File: src/main/java/edu/cornell/kfs/sys/service/impl/WebServiceCredentialServiceImpl.java [hunk 4]
```diff
@@ -16,31 +17,46 @@
...
+        keyMap.put(CUKFSPropertyConstants.WEB_SERVICE_CREDENTIAL_GROUP_CODE, credentialGroupCode);
         keyMap.put(CUKFSPropertyConstants.WEB_SERVICE_CREDENTIAL_KEY, credentialKey);
-        WebServiceCredential webServiceCredential = businessObjectService.findByPrimaryKey(WebServiceCredential.class, keyMap);
-        
-        if(ObjectUtils.isNotNull(webServiceCredential)){
-            credentialValue = webServiceCredential.getCredentialValue();
+        keyMap.put(KFSPropertyConstants.ACTIVE_INDICATOR, KFSConstants.ACTIVE_INDICATOR);
+        Collection<WebServiceCredential> webServiceCredentials = businessObjectService.findMatching(WebServiceCredential.class, keyMap);
+
+        if(webServiceCredentials.size() > 0){
```
Comment (cah292) [inline src/main/java/edu/cornell/kfs/sys/service/impl/WebServiceCredentialServiceImpl.java]: Missing spaces before the opening parentheses and before the curly brace.

File: src/main/java/edu/cornell/kfs/sys/service/impl/WebServiceCredentialServiceImpl.java [hunk 5]
```diff
@@ -16,31 +17,46 @@
...
         Map<String, String> keyMap = new HashMap<String, String>();
+        keyMap.put(CUKFSPropertyConstants.WEB_SERVICE_CREDENTIAL_GROUP_CODE, credentialGroupCode);
         keyMap.put(CUKFSPropertyConstants.WEB_SERVICE_CREDENTIAL_KEY, credentialKey);
-        WebServiceCredential webServiceCredential = businessObjectService.findByPrimaryKey(WebServiceCredential.class, keyMap);
-        
-        if(ObjectUtils.isNotNull(webServiceCredential)){
+        keyMap.put(KFSPropertyConstants.ACTIVE_INDICATOR, KFSConstants.ACTIVE_INDICATOR);
+        Collection<WebServiceCredential> webServiceCredentials = businessObjectService.findMatching(WebServiceCredential.class, keyMap);
+
+        if(webServiceCredentials.size() > 0){
```
Comment (cah292) [inline src/main/java/edu/cornell/kfs/sys/service/impl/WebServiceCredentialServiceImpl.java]: Missing spaces before the opening parentheses and before the curly brace.

File: src/main/java/edu/cornell/kfs/sys/service/impl/WebServiceCredentialServiceImpl.java [hunk 6]
```diff
@@ -16,31 +17,46 @@
...
+            WebServiceCredential webServiceCredential = webServiceCredentials.iterator().next();
             webServiceCredential.setCredentialValue(credentialValue);
             businessObjectService.save(webServiceCredential);
         }
     }
 
+    @Override
+    public Collection<WebServiceCredential> getWebServiceCredentialsByGroupCode(String credentialGroupCode){
```
Comment (cah292) [inline src/main/java/edu/cornell/kfs/sys/service/impl/WebServiceCredentialServiceImpl.java]: Missing space before the curly brace.

File: src/test/java/edu/cornell/kfs/concur/ConcurUtilsTest.java [hunk 7]
```diff
@@ -67,12 +67,12 @@ public void extractAccountNumberFromConcurValue(){
     }
     
     @Test
-    public void extracCodeFromEmptyString(){      
+    public void extractCodeFromEmptyString(){
```
Comment (cah292) [inline src/test/java/edu/cornell/kfs/concur/ConcurUtilsTest.java]: I appreciate your desire to get this unit test method corrected (and the other ones as well). However, if these methods don't have anything to do with the web service credential changes for this story, then I think these unit test fixes should be deferred to a separate user story to avoid scope creep.

**Comment by db483** [comment]
retest this please

**Comment by ajd299** [comment]
I've made some more changes to address the issues. This is now ready for review again.
